lib/bacnet-asn1.js

```diff
@@ -699,11 +700,15 @@ var bacapp_decode_application_data = module.exports.bacapp_decode_application_da
       var len = result.len;
       result = bacapp_decode_data(buffer, offset + len, maxOffset, result.tagNumber, result.value);
       if (!result) return;
-      return {
+      var resObj = {
         len: len + result.len,
         type: result.type,
         value: result.value
       };
+      if (result.encoding !== undefined) {
+        resObj.encoding = result.encoding;
```

Comment:
Do we have to do a check for undefined here? Doesn't all string have some kind of encoding?